File: src/main/java/view/Flash.java
package view;

import domain.PendingMove;
import org.jetbrains.annotations.NotNull;
import org.jline.terminal.Terminal;

class Flash {
    private String   mText;
    private Terminal mTerminal;

    // lifetime
    Flash(Terminal terminal) {
        mTerminal = terminal;
    }

    // commands
    void render(@NotNull Error error) {
        if(error instanceof PendingMove.PositionWasOccuppied) {
            mText = "The position " + ((PendingMove.PositionWasOccuppied) error).getNewPosition() + " is occupied.";
        } else if (error instanceof PendingMove.PositionOutOfBounds) {
            mText = "The position " + ((PendingMove.PositionOutOfBounds) error).getNewPosition() + " is out of bounds.";
        } else if (error instanceof PendingMove.InvalidPawnId) {
            mText = "There is no pawn with id " + ((PendingMove.InvalidPawnId) error).getPawnId();
        }
    }

    void draw() {
        if(mText != null) {
            mTerminal.writer().println();
            mTerminal.writer().println(mText);
        }

        mText = null;
    }
}

File: Sources/Onitama/view/Flash.swift
final class Flash{
  private var mText:     String?
  private let mTerminal: Terminal

  // lifetime
  init(terminal: Terminal) {
    mTerminal = terminal
  }

  // commands
  func render(error: PendingMove.Error) {
    switch error {
    case .positionWasOccupied(let position):
      mText = "The position \(position) is occupied."
    case .positionOutOfBounds(let position):
      mText = "The position \(position) is out of bounds."
    case .invalidPawnId(let id):
      mText = "There is no pawn with id \(id)."
    case .missingCard:
      mText = "The card is missing."
    case .missingPawnId:
      mText = "The pawn id is missing."
    }
  }

  func draw() {
    if let text = mText {
      mTerminal.write("\n\(text)")
    }
    
    mText = nil
  }
}
